index.bs

```diff
@@ -1923,6 +1923,39 @@ This section defines an initial set of extensions. These are recommended for imp
 interoperability.
 
 
+## FIDO AppId ## {#extension-appid}
+
+This authentication extension allows [RP]s who have previously registered an
+Authentication using the legacy FIDO JavaScript APIs to communicate with
+authenticators.  Specifically, this extension allows [RP]s to specify an
+|appId| [[FIDO-APPID]] to overwrite the otherwise computed |rpId|.  This
+extension is only valid if used during the <a>getAssertion()</a> call; other
+usage will result in client error.
+
+: Extension identifier
+:: `fido_appid`
+
+: Client argument
+:: A single UTF-8 encoded string specifying a FIDO |appId|.
+
+: Client processing
+:: Replace the calculation of |rpId| in Step 3 of [[#getAssertion]] with the
+    following procedure:  The client uses the value of |fido_appid| to perform
+    the AppId validation procedure (as defined by [[FIDO-APPID]]).  If valid,
+    the value of |rpId| for all client processing should be replaced by the
+    value of |fido_appid|.  It is an application error to both use this
```

Comment:
Might be better to make the error explicit and move it to the beginning of the procedure. Something like "If {{AssertionOptions/rpId}} is present, reject promise with a DOMException whose name is "NotAllowedError", and terminate this algorithm. Otherwise, ..."